x-pack/plugins/reporting/server/lib/tasks/harvest_reports.ts

```diff
@@ -0,0 +1,142 @@
+    this.timeout = numberToDuration(config.get('queue', 'timeout'));
+  }
+
+  private async getStore(): Promise<ReportingStore> {
+    if (this.store) {
+      return this.store;
+    }
+    const { store } = await this.reporting.getPluginStartDeps();
+    this.store = store;
+    return store;
+  }
+
+  public async init(taskManager: TaskManagerStartContract) {
+    this.taskManagerStart = taskManager;
+
+    // Round the interval up to the nearest second since Task Manager doesn't
+    // support milliseconds
+    const scheduleInterval =
+      Math.ceil(numberToDuration(this.config.get('queue', 'pollInterval')).asSeconds()) + 's';
+
+    await taskManager.ensureScheduled({
+      id: this.TYPE,
+      taskType: this.TYPE,
+      state: {},
+      params: {},
+      schedule: { interval: scheduleInterval },
+    });
+  }
+
+  private getTaskRunner(): TaskRunCreatorFunction {
+    return () => {
+      return {
+        run: async () => {
+          const reportingStore = await this.getStore();
+
+          try {
+            this.logger.debug('Checking for expired reports...');
+            const results = await reportingStore.findExpiredReports();
+            if (!results || results.length < 1) {
+              return;
+            }
+
+            for (const expired of results) {
+              const {
+                _id: jobId,
+                _source: { process_expiration: processExpiration, status },
+              } = expired;
+              const expirationTime = moment(processExpiration);
+              const timeWaitValue = moment().valueOf() - expirationTime.valueOf();
+              const timeWaitTime = moment.duration(timeWaitValue);
+              this.logger.info(
+                `Task ${jobId} has ${status} status for ${timeWaitTime.humanize()}. The queue timeout is ${this.timeout.humanize()}.`
+              );
+
+              // clear process expiration and reschedule
+              const oldReport = new Report({ ...expired, ...expired._source });
+              const reschedulingTask = oldReport.toReportTaskJSON();
+              await reportingStore.clearExpiration(oldReport);
+              await this.rescheduleTask(reschedulingTask, this.logger);
+              // TODO handle error
+              // if there is an error that is not a conflict, then mark the report failed?
+            }
+          } catch (err) {
+            this.logger.error('Could not find and update expired reports!');
+            this.logger.error(err);
+          }
+
+          return;
+        },
+
+        cancel: async () => ({ state: {} }),
+      };
+    };
+  }
+
+  public getTaskDefinition() {
+    return {
+      type: REPORTING_HARVEST_TYPE,
+      title: 'Reporting: Find expired jobs to retry',
+      createTaskRunner: this.getTaskRunner(),
+      maxAttempts: 1,
+      // round the timeout value up to the nearest second, since Task Manager
+      // doesn't support milliseconds
+      timeout: this.timeout.asSeconds() + 's',
```

Comment:
To remove the error below, I made the suggested change and it seems to have fixed the failing test.

```
[error][plugins][taskManager] Failed to mark Task reports:harvest "reports:harvest" as running: Invalid interval "121.234s". Intervals must be of the form {number}m. Example: 5m.
```

```suggestion
      timeout: Math.ceil(this.timeout.asSeconds()) + 's',
```

Reply:
Now to investigate why there's interference between the two.